pkg/actions/lua/lakefs/client.go

```diff
@@ -39,12 +41,15 @@ func newLakeFSRequest(ctx context.Context, user *model.User, method, url string,
 	if data == nil {
 		body = bytes.NewReader(data)
 	}
-	req, err := http.NewRequestWithContext(ctx, method, url, body)
+
+	// Chi stores its routing information on the request context which breaks this sub-request's routing.
+	// We explicitly nullify any existing routing information before creating the new request
+	subCtx := auth.WithUser(context.WithValue(ctx, chi.RouteCtxKey, nil), user)
+	req, err := http.NewRequestWithContext(subCtx, method, url, body)
```

Comment:
1. Suggested small re-write to make the code more clear.
2. If our context already passed the auth middleware - the user already set, right?
```suggestion
	// Chi stores its routing information on the request context which breaks this sub-request's routing.
	// We explicitly nullify any existing routing information before creating the new request
	ctx = context.WithValue(ctx, chi.RouteCtxKey, nil)
	// Add user to the request context
	ctx = auth.WithUser(ctx, user)
	req, err := http.NewRequestWithContext(ctx, method, url, body)
```

Reply:
Thanks, modified.
I prefer to explicitly set the given user - in cases where we decide to execute a hook without being triggered by another HTTP request, this assumption will break. 

I prefer to treat context as an opaque value without assuming anything about it (except of course, to clean any Chi related things that will break this execution)